saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -755,18 +775,26 @@ class Meta:
         error_type_field = "stock_errors"
 
     @classmethod
+    @traced_atomic_transaction()
     def perform_mutation(cls, root, info, **data):
+        manager = info.context.plugins
         variant = cls.get_node_or_error(
             info, data["variant_id"], only_type=ProductVariant
         )
         warehouses_pks = cls.get_global_ids_or_error(
             data["warehouse_ids"], Warehouse, field="warehouse_ids"
         )
-        warehouse_models.Stock.objects.filter(
+        stocks_to_delete = warehouse_models.Stock.objects.filter(
             product_variant=variant, warehouse__pk__in=warehouses_pks
-        ).delete()
+        )
 
         variant = ChannelContext(node=variant, channel_slug=None)
+
+        for stock in stocks_to_delete:
+            trigger_out_of_stock_webhook(stock, manager)
```

Comment:
We should also check that we receive a proper payload in delete mutation. If you generate payload after you remove stock. `stock.id` for this object will be `None`